src/Symfony/Component/Mime/Tests/MessageTest.php

```diff
@@ -125,6 +125,14 @@ public function testGetPreparedHeadersHasSenderWhenNeeded()
         $this->assertEquals('thomas@symfony.com', $message->getPreparedHeaders()->get('Sender')->getAddress()->getAddress());
     }
 
+    public function testGenerateMessageIdThrowsWhenHasFromButNoAddresses()
+    {
+        $this->expectException(\LogicException::class);
```

Comment:
please move this line right before the call of `generateMessageId()`, also we should change the class name to actually check for the `LogicException` that is thrown in the `Message` class (i.e. the one from the Mime component)